stdlib.sh

```diff
@@ -19,7 +19,16 @@ direnv="$(command -v direnv)"
 DIRENV_LOG_FORMAT="${DIRENV_LOG_FORMAT-direnv: %s}"
 
 # Where direnv configuration should be stored
-direnv_config_dir="${DIRENV_CONFIG:-${XDG_CONFIG_HOME:-$HOME/.config}/direnv}"
+IFS=: xdg_config_dirs=( ${XDG_CONFIG_DIRS:-/etc/xdg} )
```

Comment:
```suggestion
direnv_config_dir="${DIRENV_CONFIG:-${XDG_CONFIG_HOME:-$HOME/.config}/direnv}"
IFS=: xdg_data_dirs=( ${XDG_DATA_DIRS:-/usr/local/share/:/usr/share} )
```
I find the XDG spec confusing. I think in this scenario, the `~/.config/direnv/lib/*.sh` can be considered "user" configuration, but the shared libraries are probably part of XDG_DATA_DIRS? Or it should have been XDG_DATA_HOME already.

Reply:
The spec says this:

> `$XDG_DATA_HOME` defines the base directory relative to which user-specific data files should be stored

It doesn't explain what "user-specific data files" are, but from my experience I'd say it's "files that contain persistent user-specific data but which are created and managed by the program". I don't think such direnv libraries fit in there, since they're neither user-specific, nor created or managed by the program. And `$XDG_DATA_DIRS` is just the extension of `$XDG_DATA_HOME` to multiple additional directories.

I really think `$XDG_CONFIG_HOME` (and its extension to `$XDG_CONFIG_DIRS`) is the right place for these libraries, they really are configuration: They don't store user-specific data, are reusable between arbitrarily many installation, they're kind of like plugins.

Reply:
I agree for `~/.config/direnv/lib/*.sh`, but as soon as you start scanning outside of the user's home, it becomes re-distributable code. For nix-direnv, you wouldn't expect the user to change the implementation themselves but rather to propose the changes upstream.

This also shows this: https://github.com/NixOS/nixpkgs/pull/192667#discussion_r980135486

Reply:
I don't understand, are you saying that because libraries outside `~` are redistributable (though I'd argue libraries in `~` are equally as redestributable), `XDG_DATA_DIRS` is the correct variable to read from? That doesn't make sense to me, `XDG_DATA_DIRS` like `XDG_DATA_HOME` is for data, not configuration or libraries or redistributable things. For GDPR compliance for example, if somebody asks me to wipe all user data, I'd wipe all `XDG_DATA_DIRS` and `XDG_DATA_HOME`, the applications should all continue working with the same configuration though